src/Microsoft.IdentityModel.Tokens/LogMessages.cs

```diff
@@ -105,6 +105,7 @@ internal static class LogMessages
         public const string IDX10509 = "IDX10509: Signature validation failed. The user defined 'Delegate' specified in TokenValidationParameters did not return a '{0}', but returned a '{1}' when reading token: '{2}'.";
         public const string IDX10510 = "IDX10510: Signature validation failed. The user defined 'Delegate' specified in TokenValidationParameters returned null when reading token: '{0}'.";
         public const string IDX10511 = "IDX10511: Signature validation failed. Keys tried: '{0}'. \nkid: '{1}'. \nExceptions caught:\n '{2}'.\ntoken: '{3}'.";
```

Comment:
Change 'kid' to 'KeyInfo'

Reply:
IDX10511 is not used for saml. For IDX10501 I added IDX10513 to be saml specific.

Reply:
Looks like 10503 should have kid -> KeyInfo then?

Reply:
10503  is used in M.IM.JsonWebTokens and S.IM.T.jwt, saml specific for 10503 is 10512.